lua/nvim-treesitter/ts_utils.lua

```diff
@@ -222,24 +227,19 @@ function M.node_length(node)
 end
 
 --- Determines whether (line, col) position is in node range
--- @param node Node defining the range
--- @param line A line (0-based)
+-- @param node TSNode defining the range
+-- @param row A row (0-based)
 -- @param col A column (0-based)
-function M.is_in_node_range(node, line, col)
-  local start_line, start_col, end_line, end_col = node:range()
-  if line >= start_line and line <= end_line then
-    if line == start_line and line == end_line then
-      return col >= start_col and col < end_col
-    elseif line == start_line then
-      return col >= start_col
-    elseif line == end_line then
-      return col < end_col
-    else
-      return true
-    end
-  else
-    return false
-  end
+function M.is_in_node_range(node, row, col)
+  local start_row, start_col, end_row, end_col = node:range()
```

Comment:
That's interesting idea, but i don't feel confident that it will not break something.

I left `is_in_node_range` unchanged, but added `is_node_end` specifically to catch one case which fixes bug with python.

@vigoux what do you think about this version of PR, and about change proposed by @stsewd ?